projects/step-frontend/src/lib/plugins-initializer/plugins-initializer.ts

```diff
@@ -66,6 +68,19 @@ const loadPlugins = (compiler: Compiler, injector: Injector) => {
       }
     );
 
+    const pluginNames = [
+      ...legacy.reduce((res, pluginInfo) => [...res, ...pluginInfo.angularModules], [] as string[]),
+      ...microfrontend.map((pluginInfo) => pluginInfo.name),
+    ];
+    registry.register(...pluginNames);
+
+    legacy = legacy.filter((l) => {
+      const ignoredPlugins = IGNORE_PLUGINS.filter((name) => l.angularModules.includes(name));
+      return ignoredPlugins.length === 0;
+    });
+
+    microfrontend = microfrontend.filter((m) => !IGNORE_PLUGINS.includes(m.name));
```

Comment:
I'm not entirely sure what happens here. Is this the part that solves my question earlier about how to manage which plugins to enable?

If yes, can we change the code so that it uses ENABLED_PLUGINS instead of IGNORED_PLUGINS so it is more clear what happens here?

Reply:
I don't think that such names will describe exactly what's happened here.
In lines 71 - 75, we collect the names of **all** plugin's modules that we were able to receive. Then we are putting it to PluginRegistry service.
PluginRegistry is just a simple storage - later StepEnterpriseCore will use it to determine, which part should be enabled.

Then as backend returns some old data (like link to multienancy), we need to exclude it, to prevent it's initialization as it has become a part of StepEnterpriseCore. That was the INGNORE_PLUGINS do.
In future as all plugins will be moved to StepEnterpriseCore and rest endpoint will be updated, INGNORE_PLUGINS should be removed from the code